blockscout-ens/bens-logic/src/protocols/protocoler.rs

```diff
@@ -203,6 +205,13 @@ impl Protocoler {
         })
     }
 
+    pub fn extract_tld(&self, domain: &str) -> Result<Tld, ProtocolError> {
+        Tld::from_domain_name(domain).ok_or_else(|| ProtocolError::InvalidName {
+            name: domain.to_string(),
+            reason: "Invalid TLD".to_string(),
+        })
+    }
+
```

Comment:
this function doesnt use self at all. also, you dont use it anywhere else except this file, so `pub` is not needed.

move this function to the end of file and make private to highlight that you use it only here